RustFxA/RustFirefoxAccounts.swift

```diff
@@ -50,15 +50,21 @@ open class RustFirefoxAccounts {
      hook into notifications like `.accountProfileUpdate` to refresh once initialize() is complete.
      Or they can wait on the accountManager deferred to fill.
      */
-    public static func startup(prefs _prefs: Prefs) -> Deferred<FxAccountManager> {
-        prefs = _prefs
-        if let manager = RustFirefoxAccounts.shared.accountManager.peek() {
-            return Deferred(value: manager)
+    public static func startup(prefs: Prefs) -> Deferred<FxAccountManager> {
+        assert(Thread.isMainThread)
```

Comment:
All firefox iOS code is main-thread only unless specified otherwise, however, I want to catch any future accidental misuse of this, it would cause subtle bugs I wouldn't want to track down.